### DISCUSSION TITLE
Zephyr connection manager - bridge multiple interfaces?

### DISCUSSION THREAD

**Discussion by kvanbere** [zephyrproject-rtos/zephyr#75852] [Q&A]
I am reading through the source code in the NRF Connect sample here, which is based on Zephyr:
https://github.com/nrfconnect/sdk-nrf/tree/main/samples/net/mqtt

The connection manager docs mention this:
```
Applications can also use Connection Manager to generically manage and use multiple connectivity technologies **simultaneously**.
```
https://docs.zephyrproject.org/latest/connectivity/networking/conn_mgr/main.html#connectivity-control

However I have a few questions.

1. How does the MQTT sample code know to use the connection manager? I was unable to find any link between the MQTT part and the connection manager itself.

2. Does the use-case of LTE + WiFi on the same board simultaneously currently work? The sample seems to imply that only one or the other works, but the Zephyr docs imply that interfaces may work simultaneously.

3. How does Zephyr track priority of network interfaces when making connections. For example, if both LTE and WiFi are available, how does it know to use WiFi instead of pushing MQTT over a metered network? Is this configurable? Where are the docs?

4. Does the connection manager support a 3-way or N-ways fallback scenario, for example Ethernet > WiFi > Cellular.

5. Filtering interfaces based on actual connectivity to the internet, perhaps using a ping server - how? Example use case would be to fall back to Cellular if the WiFi connection stops being able to connect to the internet for some reason. This would allow the device to receive MQTT updates for new WiFi configurations rather than getting bricked.

I know it's a lot of questions but if you know answers to any one question or all the questions would love to hear from you.

Thanks. 😄

**Comment by kvanbere** [comment]
Hm, is this up to date?
https://github.com/zephyrproject-rtos/zephyr/blob/fd0e46506fabd4c0c91ecb336a39cb6a6aa2b270/doc/connectivity/networking/overview.rst#id27

Maybe relevant
```
Multiple Network Technologies.
The Zephyr OS can be configured to support multiple network technologies at the same
time simply by enabling them in Kconfig: for example, Ethernet, Wi-Fi and 802.15.4 support.
Note that no automatic IP routing functionality is provided between these technologies.
Applications can send data according to their needs to desired network interface.
```

Re: "Applications can send data according to their needs to desired network interface" -- how does this work, I don't see code for this in the MQTT example?

Edit:
I did some digging and the MQTT uses the zsocks BSD compatible API, which seems to select the appropriate interface based on the IPv routing table. I have no idea how this works when there are 2 internet-connected interfaces active. Perhaps I am supposed to make some kind of software bridge device?

**Reply by dheadrick1618** [reply]
Did you  ever resolve this? IIRC the mqtt_client struct has an 'if_name' field that you can pass the name of the network interface you want it to use. However I am not sure if it existed when you made this post.

**Comment by kvanbere** [comment]
Bump.
